packages/blade/src/components/DatePicker/Calendar.web.tsx

```diff
@@ -0,0 +1,172 @@
+      });
+    }
+  }
+
+  const [_date, setDate] = useUncontrolledDates({
+    type: 'default',
+    value: date,
+    defaultValue: defaultDate,
+    onChange: onDateChange as any,
+  });
+
+  const ctx = useDatesContext();
+  const isMobile = useIsMobile();
+  const currentDate = _date || shiftTimezone('add', new Date(), ctx.getTimezone());
+  const numberOfColumns = isMobile ? 1 : isRange ? 2 : 1;
+  const columnsToScroll = numberOfColumns || 1;
+
+  const handleNextMonth = () => {
+    const nextDate = dayjs(currentDate).add(columnsToScroll, 'month').toDate();
+    onNextMonth?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handlePreviousMonth = () => {
+    const nextDate = dayjs(currentDate).subtract(columnsToScroll, 'month').toDate();
+    onPreviousMonth?.(nextDate);
+    onPrevious?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handleNextYear = () => {
+    const nextDate = dayjs(currentDate).add(columnsToScroll, 'year').toDate();
+    onNextYear?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handlePreviousYear = () => {
+    const nextDate = dayjs(currentDate).subtract(columnsToScroll, 'year').toDate();
+    onPreviousYear?.(nextDate);
+    onPrevious?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handleNextDecade = () => {
+    const nextDate = dayjs(currentDate)
+      .add(10 * columnsToScroll, 'year')
+      .toDate();
+    onNextDecade?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  const handlePreviousDecade = () => {
+    const nextDate = dayjs(currentDate)
+      .subtract(10 * columnsToScroll, 'year')
+      .toDate();
+    onPreviousDecade?.(nextDate);
+    onNext?.(nextDate);
+    setDate(nextDate);
+  };
+
+  return (
+    <CalendarStyles
+      display="flex"
+      flexDirection="column"
+      gap="spacing.7"
+      pickerType={levelToPicker[level]}
+    >
+      <CalendarHeader
+        isRange={isRange}
+        date={currentDate}
+        onLevelChange={(level) => setLevel(() => level)}
+        pickerType={levelToPicker[level] as PickerType}
+        onNextMonth={handleNextMonth}
+        onPreviousMonth={handlePreviousMonth}
+        onNextDecade={handleNextDecade}
+        onPreviousDecade={handlePreviousDecade}
+        onNextYear={handleNextYear}
+        onPreviousYear={handlePreviousYear}
+      />
+      <CalendarGradientStyles isRange={isRange} date={currentDate}>
+        <DatePicker
+          withCellSpacing={false}
+          type={isRange ? 'range' : 'default'}
+          date={_date}
+          locale={locale}
+          level={level}
+          onDateChange={setDate}
+          onLevelChange={(level) => setLevel(() => level)}
+          numberOfColumns={numberOfColumns}
+          weekdayFormat="ddd"
+          firstDayOfWeek={firstDayOfWeek}
+          // @ts-expect-error unable to narrow props based on `type`
+          allowSingleDateInRange={allowSingleDateInRange}
+          classNames={{
```

Comment:
What are these classes?

Reply:
Mantine lets us override these classes so we can style their internal components. 

https://mantine.dev/dates/date-picker/?t=styles-api